src/hooks/useTreasury.ts

```diff
@@ -66,6 +76,211 @@ const useTreasury = () => {
     return yUsdBalance ? yUsdBalance + yamYUsdValue : yamYUsdValue;
   }, [yamBalance, yamTwap, yUsdBalance]);
 
+  const getAssetsHistory = async () => {
+    const sushiPrice = await getSUSHIPrice() || 0;
+    const wethPrice = await getWETHPrice() || 0;
+    const dpiPrice = await getDPIPrice() || 0;
+    const indexCoopPrice = await getINDEXCOOPPrice() || 0;
+    const rewardsIndexCoop = await getIndexCoopLPRewards(yam) || 0;
+    const yusdPrice = await getYUSDPrice() || 0;
+    const umaPrice = await getUMAPrice() || 0;
+    const rewardsSushi = (await getSushiRewards(yam)) || 0;
+
+    const history = {
+      WETH: {
+        11133885: {
+          value: 0,
+          tx: ''
+        },
+        11160087: {
+          value: 0,
+          tx: ''
+        },
+        11243912: {
+          value: 201 * 468,
+          tx: ''
+        },
+        11244494: {
+          value: 555 * 464,
+          tx: ''
+        },
+        11289910: {
+          value: 201 * 480,
+          tx: ''
+        },
+        11680240: {
+          value: (283 + 4.47) * 1258,
+          tx: ''
+        },
+        latest: totalWETHValue * wethPrice
+      },
+      yUSD: {
+        11133885: {
+          value: 1.15 * (1896995 + 215518 + 184500),
+          tx: ''
+        },
+        11160087: {
+          value: 1.15 * (1896995 + 215518),
+          tx: ''
+        },
+        11243912: {
+          value: 1.18 * 1896995,
+          tx: ''
+        },
+        11244494: {
+          value: 1.18 * 1896995,
+          tx: ''
+        },
+        11289910: {
+          value: 1.19 * 1896995,
+          tx: ''
+        },
+        11680240: {
+          value: 1.19 * 1896995,
+          tx: ''
+        },
+        latest: yusdPrice * totalYUsdValue
+      },
+      DPI: {
+        11133885: {
+          value: 3351 * 75,
+          tx: ''
+        },
+        11160087: {
+          value: 3351 * 72,
+          tx: ''
+        },
+        11243912: {
+          value: 434 * 90,
+          tx: ''
+        },
+        11244494: {
+          value: 3351 * 80,
+          tx: ''
+        },
+        11289910: {
+          value: 434 * 104,
+          tx: ''
+        },
+        11680240: {
+          value: 434 * 104,
+          tx: ''
+        },
+        latest: (totalDPIValue || 0) * dpiPrice
+      },
+      INDEX: {
+        11133885: {
+          value: 0,
+          tx: ''
+        },
+        11160087: {
+          value: 0,
+          tx: ''
+        },
+        11243912: {
+          value: 0,
+          tx: ''
+        },
+        11244494: {
+          value: 0,
+          tx: ''
+        },
+        11289910: {
+          value: 36 * 11.6,
+          tx: ''
+        },
```

Comment:
for empty values you can add a condition to avoid the need of setting these each time, for any asset to come